charts/partials/templates/_proxy.tpl

```diff
@@ -9,7 +18,7 @@ env:
 - name: LINKERD2_PROXY_LOG_FORMAT
   value: {{.Values.global.proxy.logFormat | quote}}
 - name: LINKERD2_PROXY_DESTINATION_SVC_ADDR
-  value: {{ternary "localhost.:8086" (printf "linkerd-dst-headless.%s.svc.%s:8086" .Values.global.namespace .Values.global.clusterDomain) (eq .Values.global.proxy.component "linkerd-destination")}}
+  value: {{ternary "localhost.:8086" (printf "linkerd-dst-headless.%s.svc.%s:8086" .Values.global.namespace .Values.global.clusterDomain) $isDestination }}
```

Comment:
One way of avoiding the new boilerplate at the beginning of this file while avoiding an error when using `eq` to compare against a potentially nil value is to do this:
```
value: {{ternary "localhost.:8086" (printf "linkerd-dst-headless.%s.svc.%s:8086" .Values.global.namespace .Values.global.clusterDomain) (eq (default .Values.global.proxy.component false) "linkerd-destination")}}
```

Reply:
Thanks for the suggestion! It seems to work with a slight chance that we discussed! :+1: